File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizRibbonTab.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * BizRibbonTab - Jetpack Compose component matching Web UI-Framework `PPRibbonTab.vue`.
 */
@Composable
fun BizRibbonTab(
    modifier: Modifier = Modifier,
    title: String = "",
    color: String = "",
    id: String = "",
    onClick: () -> Unit = {},
    content: @Composable (() -> Unit)? = null
) {
    val primaryNavy = Color(0xFF1A2A5E)
    val primaryBlue = Color(0xFF003399)
    val secondaryBg = Color(0xFFF1F5F9)
    val borderColor = Color(0xFFCBD5E1)

    Box(
        modifier = modifier
            .fillMaxWidth()
            .defaultMinSize(minHeight = 44.dp)
            .background(Color.White, shape = RoundedCornerShape(12.dp))
            .border(1.dp, borderColor, shape = RoundedCornerShape(12.dp))
            .padding(16.dp)
            .clickable(onClick = onClick),
        contentAlignment = Alignment.CenterStart
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween,
            modifier = Modifier.fillMaxWidth()
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = "RibbonTab",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = primaryNavy
                )
                if (content != null) {
                    Spacer(modifier = Modifier.height(8.dp))
                    content()
                }
            }
        }
    }
}

/**
 * Web UI-Framework parity alias for [BizRibbonTab].
 */
@Composable
fun PPRibbonTab(
    modifier: Modifier = Modifier,
    title: String = "",
    color: String = "",
    id: String = "",
    onClick: () -> Unit = {},
    content: @Composable (() -> Unit)? = null
) = BizRibbonTab(
        modifier = modifier,
        title = title,
        color = color,
        id = id,
        onClick = onClick,
        content = content
)

File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizRibbonTab.swift
import SwiftUI

/**
 * BizRibbonTab - SwiftUI component matching Web UI-Framework `PPRibbonTab.vue`.
 */
public struct BizRibbonTab: View {
    public var title: String
    public var color: String
    public var id: String
    public var action: () -> Void

    public init(
        title: String = "",
        color: String = "",
        id: String = "",
        action: @escaping () -> Void = {}
    ) {
        self.title = title
        self.color = color
        self.id = id
        self.action = action
    }

    public var body: some View {
        Button(action: action) {
            VStack(alignment: .leading, spacing: 8) {
                Text("RibbonTab")
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(Color(hex: "#1A2A5E"))
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(16)
            .background(Color.white)
            .cornerRadius(12)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(Color(hex: "#CBD5E1"), lineWidth: 1)
            )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

/**
 * Web UI-Framework parity alias for [BizRibbonTab].
 */
public typealias PPRibbonTab = BizRibbonTab
